lib/api/src/instance.rs

```diff
@@ -85,11 +124,21 @@ impl Instance {
             })
             .collect::<Exports>();
 
-        Ok(Self {
+        let instance = Self {
             handle: Arc::new(Mutex::new(handle)),
             module: module.clone(),
             exports,
-        })
+        };
+
+        unsafe {
+            instance
+                .handle
+                .lock()
+                .unwrap()
```

Comment:
Use expect rather than unwrap

Reply:
`unwrap` on `Mutex::lock` isn't worth handling, there's nothing useful we can provide to users here

Reply:
To clarify this is only possible if there was a panic on another thread while holding the mutex.  If that happens, we just want to terminate Wasmer as soon as possible